File: gripql/python/gripql/operators.py
from __future__ import absolute_import, print_function, unicode_literals


def and_(*expressions):
    return {"and": {"expressions": expressions}}


def or_(*expressions):
    return {"or": {"expressions": expressions}}


def not_(expression):
    return {"not": expression}


def eq(key, value):
    return {"condition": {"key": key, "value": value, "condition": "EQ"}}


def neq(key, value):
    return {"condition": {"key": key, "value": value, "condition": "NEQ"}}


def gt(key, value):
    return {"condition": {"key": key, "value": value, "condition": "GT"}}


def gte(key, value):
    return {"condition": {"key": key, "value": value, "condition": "GTE"}}


def lt(key, value):
    return {"condition": {"key": key, "value": value, "condition": "LT"}}


def lte(key, value):
    return {"condition": {"key": key, "value": value, "condition": "LTE"}}


def inside(key, lower, upper):
    return {"condition": {"key": key, "value": [lower, upper], "condition": "INSIDE"}}


def outside(key, lower, upper):
    return {"condition": {"key": key, "value": [lower, upper], "condition": "OUTSIDE"}}


def between(key, lower, upper):
    return {"condition": {"key": key, "value": [lower, upper], "condition": "BETWEEN"}}


def within(key, values):
    if not isinstance(values, list):
        if not isinstance(values, dict):
            values = [values]
    return {"condition": {"key": key, "value": values, "condition": "WITHIN"}}


def without(key, values):
    if not isinstance(values, list):
        if not isinstance(values, dict):
            values = [values]
    return {"condition": {"key": key, "value": values, "condition": "WITHOUT"}}


def contains(key, value):
    return {"condition": {"key": key, "value": value, "condition": "CONTAINS"}}

File: gripql/R/R/operators.R
#' @export
eq <- function(k, v) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = v, "condition" = "EQ"))
}

#' @export
neq <- function(k, v) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = v, "condition" = "NEQ"))
}

#' @export
gt <- function(k, v) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = v, "condition" = "GT"))
}

#' @export
gte <- function(k, v) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = v, "condition" = "GTE"))
}

#' @export
lt <- function(k, v) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = v, "condition" = "LT"))
}

#' @export
lte <- function(k, v) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = v, "condition" = "LTE"))
}

#' @export
between <- function(lower, upper) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = c(lower, upper), "condition" = "BETWEEN"))
}

#' @export
inside <- function(lower, upper) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = c(lower, upper), "condition" = "INSIDE"))
}

#' @export
outside <- function(lower, upper) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = c(lower, upper), "condition" = "OUTSIDE"))
}

#' @export
within <- function(values) {
  check_class(k, "character")
  if (length(values) == 0) {
    values <- list()
  }
  if (length(values) == 1) {
    values <- list(values)
  }
  list("condition" = list("key" = k, "value" = values, "condition" = "WITHIN"))
}

#' @export
without <- function(values) {
  check_class(k, "character")
  if (length(values) == 0) {
    values <- list()
  }
  if (length(values) == 1) {
    values <- list(values)
  }
  list("condition" = list("key" = k, "value" = values, "condition" = "WITHOUT"))
}

#' @export
contains <- function(k, v) {
  check_class(k, "character")
  list("condition" = list("key" = k, "value" = v, "condition" = "CONTAINS"))
}
